src/build.rs

```diff
@@ -998,6 +997,58 @@ fn deprecation_warnings(ws: &Workspace, args: &ArgMatches) -> anyhow::Result<()>
     Ok(())
 }
 
+/// What library types to build
+#[derive(Debug, Clone, Copy)]
+pub struct LibraryTypes {
+    pub staticlib: bool,
+    pub cdylib: bool,
+}
+
+impl LibraryTypes {
+    fn from_target(target: &target::Target) -> Self {
+        // for os == "none", cdylib does not make sense. By default cdylib is also not built on
+        // musl, but that can be overriden by the user. That is useful when musl is being used as
+        // main libc, e.g. in Alpine, Gentoo and OpenWRT
+        Self {
```

Comment:
You may link https://github.com/lu-zero/cargo-c/issues/180 or the README section, I wish there is a better way to see if the rustc in use defaults one way or the other w/out trying beforehand :/